### PR TITLE
TQ42-1558 | TQ42-1559 | TQ42-1560: Basic channel implementation

### PR DESCRIPTION
**Pull request by sbtq**

### CODE REVIEW

File: tq42/channel.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                async for incoming in timed_stream:
+                    logging.debug(f"User received {incoming=}")
+                    if incoming.HasField("ask_data"):
+                        ask_msg: ChannelMessage = incoming
+                        ack_msg = ChannelMessage(
+                            acknowledge_data=DataAcknowledge(
+                                id=incoming.sequential_message_id
+                            )
+                        )
+                        tell = callback(incoming.ask_data)
```
Comment (epetq) [inline tq42/channel.py:87]: allow callbacks be an async function too?

File: tq42/channel.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+if TYPE_CHECKING:
+    from tq42.client import TQ42Client
+
+
+from tq42.utils.timers import AsyncTimedIterable
+
+
+class Channel:
+    """
+    Class to create and view datasets
```
Comment (epetq) [inline tq42/channel.py:33]: old docs?

File: tq42/channel.py (hunk 1 above)
Comment (sbtq) [reply tq42/channel.py]: Good suggestion, adaptad 👍 ✔️

File: tq42/channel.py (hunk 2 above)
Comment (sbtq) [reply tq42/channel.py]: good catch

**Review by bdefeyter-tq — APPROVED** [review]
LGTM

**Pull request merged**